File: frontend/src/lib/schemas.ts
import { z } from "zod";

export const RecommendationSchema = z.object({
  symbol: z.string(),
  rating: z.string(),
  rationale: z.string().optional(),
});

export const InsightSchema = z.object({
  title: z.string(),
  description: z.string(),
});

File: backend/app/schemas.py
from pydantic import BaseModel
from typing import Optional


class StockQuote(BaseModel):
    symbol: str
    price: float
    change: float
    percent_change: float


class Recommendation(BaseModel):
    symbol: str
    rating: str
    rationale: Optional[str] = None


class Insight(BaseModel):
    title: str
    description: str


